idl2library.cmake

```diff
@@ -0,0 +1,226 @@
+#
+# Example:
+#           idl2library(IDLS idl/example.idl idl/std_doc.idl)
+#           idl2library(IDLS idl/example.idl idl/std_doc.idl BUILD_STATIC)
+#
+
+cmake_minimum_required(VERSION 3.20)
+
+set (IDL_TARGET_SUFFIX "_idl")
+
+#Converts idl file to a target name:  lowervase with no path or extension with IDL_TARGET_SUFFIX appended
+macro(get_idl_target_name idlfile output)
+    get_filename_component(current_idl_shortname ${idlfile} NAME_WE)
+    string(TOLOWER "${current_idl_shortname}" current_idl_shortname)
+    set(temp_output "${current_idl_shortname}")
+    string(APPEND temp_output "${IDL_TARGET_SUFFIX}")
+    set(${output} ${temp_output})
+endmacro()
+
+#Finds all of the input file's (absolute path) dependencies and returns them in IDL_TARGET_DEPENDENCIES
+function(find_idl_dependencies input_file)	
+	unset(search_list)
+	unset(return_list)
+	LIST(APPEND current_list ${input_file}) #Files that need to be searched (full path)
+	LIST(APPEND search_list ${input_file}) #Don't search more than once (full path)
+	
+	while(current_list)
+	    SET(file_list ${current_list})
+		unset(current_list)
+		foreach(current_file ${file_list})
+			set (INCLUDE_REGEX "^[ \t]*\#include[ \t]+\"(.+\\.[I|i][D|d][L|l])\".*$")
+			file(STRINGS ${current_file} included_files REGEX ${INCLUDE_REGEX})
+			foreach(included_file ${included_files})
+				string(REGEX REPLACE ${INCLUDE_REGEX} "\\1" included_filename "${included_file}")
+				get_idl_target_name(${included_filename} current_include_target)
+				if (NOT ${${current_include_target}_ABSPATH} IN_LIST search_list)
+					LIST(APPEND current_list ${${current_include_target}_ABSPATH})
+					LIST(APPEND return_list ${current_include_target})
+					LIST(APPEND search_list ${${current_include_target}_ABSPATH}) #The file should now never be searched again
+				endif()
+			endforeach()
+		endforeach()
+		unset(file_list)
+	endwhile()
+	set(IDL_TARGET_DEPENDENCIES ${return_list} PARENT_SCOPE)
+endfunction()
+
+function(idl2library)
+    set(options BUILD_STATIC)
+	set(multiValueArgs IDLS)
+	
+	cmake_parse_arguments(idl2library "BUILD_STATIC" "" "IDLS" ${ARGN})
+
+    if (${idl2library_BUILD_STATIC})
+	    message("Configured to build static libraries.")
+	else()
+	    message("Configured to build shared libraries.")
+	endif()
+
+    unset(IDL_WISHLIST)
+	#Add idl excention to each file in the list if there is no extension
+    foreach(IDL_ARG ${idl2library_IDLS})
+        string(REGEX MATCH "\.[I|i][D|d][L|l]$" IDL_ARG_EXTENSION ${IDL_ARG})
+        if(IDL_ARG_EXTENSION STREQUAL "")
+            #If they didn't then add it..
+            set(IDL_ARG "${IDL_ARG}.idl")
+        endif()
+        list(APPEND IDL_WISHLIST ${IDL_ARG})
+    endforeach()
+    
+    option(OPENDDS_CPP11_IDL_MAPPING "Use C++11 IDL mapping" OFF)	
+    option(OPENDDS_CMAKE_VERBOSE "Print verbose output when loading the OpenDDS Config Package" ON)
+    option(OPENDDS_ALLOW_ENV_CHANGE "Projects may reset OpenDDS env vars." ON)
```

Comment:
```suggestion
```

This hasn't done anything for a long time.

Reply:
Removed.